src/framework/entity.js

```diff
@@ -605,8 +609,6 @@ class Entity extends GraphNode {
     }
 
     _getSortedComponents() {
-        _sortedArray.length = 0;
```

Comment:
I'd add `Debug.assert(_sortedArray.length === 0);` to validate, as it's a bit of an easy to break concept.